### PR TITLE
Improve ts support

### PR DESCRIPTION
**Pull request by interstella-5555**
- document `this` with typescript on client methods
- clean up `CustomClientOptions`
- replace `windows.location = url` with `window.location.assign(url)` to make typescript and tests happy

TODO: @m-cat will update docs on client options

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -17,25 +17,17 @@ import { addUrlQuery, defaultPortalUrl, makeUrl } from "./utils";
 import { getFile, setFile } from "./skydb";
 import { lookupRegistry, updateRegistry } from "./registry";
 
+export type CustomClientOptions = {
+  APIKey?: string; // authentication password to use
```
Comment (mrcnski) [inline src/client.ts]: Can we use JSDoc to document these parameters?

File: src/client.ts [hunk 2]
```diff
@@ -17,25 +17,17 @@ import { addUrlQuery, defaultPortalUrl, makeUrl } from "./utils";
...
+
 export class SkynetClient {
   portalUrl: string;
-  customOptions: Record<string, unknown>;
+  customOptions: CustomClientOptions;
 
-  /**
-   * The Skynet Client which can be used to access Skynet.
-   * @constructor
-   * @param {string} [portalUrl="https://siasky.net"] - The portal URL to use to access Skynet, if specified. To use the default portal while passing custom options, use "".
```
Comment (mrcnski) [inline src/client.ts:27]: Why remove all this documentation instead of just the irrelevant parameters?

**Review by mrcnski — APPROVED** [review]

**Review by mrcnski — CHANGES_REQUESTED** [review]
Changes requested!

File: src/client.ts (hunk 2 above)
Comment (interstella-5555) [reply src/client.ts:27]: yeah sorry, my bad

File: src/client.ts (hunk 1 above)
Comment (interstella-5555) [reply src/client.ts]: jsdoc doesn't apply to typescript

Comment (interstella-5555) [reply src/client.ts]: replaced `// comment` with `/** comment */` so they are preserved in the resulting d.ts file

File: src/client.ts (hunk 2 above)
Comment (mrcnski) [reply src/client.ts:27]: Should we indicate that these parameters are optional? (e.g. with brackets in the JSDoc or a `?` after the parameter name)

Comment (interstella-5555) [reply src/client.ts:27]: yeah, wrapped back in brackets

**Review by mrcnski — APPROVED** [review]

**Pull request merged**